Source/WebKit/UIProcess/Cocoa/UIDelegate.mm

```diff
@@ -448,6 +449,17 @@
         return;
     }
 
+    if (m_uiDelegate->m_delegateMethods.webViewRequestStorageAccessPanelWithoutQuirksUnderCurrentDomainCompletionHandler) {
```

Comment:
I think the new delegate method should be only handling quirk cases (instead of skipping the quirk check), so that when client specifies both methods, one of the delegate methods will be invoked (in the existing approach, only `webViewRequestStorageAccessPanelWithoutQuirksUnderCurrentDomainCompletionHandler` will be invoked).
Something like:
```
if (organizationStorageAccessPromptQuirk || additionalLoginDomain) {
    if (m_uiDelegate->m_delegateMethods.webViewRequestStorageAccessPanelFromQuirksUnderCurrentDomainCompletionHandler) {
         // ...
    }
}
```
And our test will be able to confirm storage access request is from quirks by providing both delegate methods.

Reply:
Okay, sounds good! I'll make that change.